src/coreclr/gc/handletable.cpp

```diff
@@ -610,6 +601,18 @@ void HndWriteBarrier(OBJECTHANDLE handle, OBJECTREF objref)
         }
     }
 }
+
+void HndWriteBarrier(OBJECTHANDLE handle, OBJECTREF objref)
```

Comment:
Would this split regress performance of the handle setters? Does this need to be inlineable in the header?

Reply:
I don't believe so. since now there's very little left in HndWriteBarrier (in ret) it'll get inlined instead. and if I look at it it does appear that way -

```
0:000> x coreclr!*hndwritebarrier*
*** WARNING: Unable to verify checksum for C:\runtime-reg\artifacts\tests\coreclr\windows.x64.Release\Tests\Core_Root\CoreCLR.dll
00007ff8`21d5d3d0 CoreCLR!HndWriteBarrierWorker (struct OBJECTHANDLE__ *, class Object *)
00007ff8`21c98cda CoreCLR!HndWriteBarrier =  (inline caller) CoreCLR!HndAssignHandle+1a
00007ff8`21c98a89 CoreCLR!HndWriteBarrier =  (inline caller) CoreCLR!SetDependentHandleSecondary+15
00007ff8`21c98e1c CoreCLR!HndWriteBarrier =  (inline caller) CoreCLR!GCHandleStore::CreateHandleOfType+bc
00007ff8`21c98d1d CoreCLR!HndWriteBarrier =  (inline caller) CoreCLR!GCHandleManager::InterlockedCompareExchangeObjectInHandle+1d
00007ff8`21f1c556 CoreCLR!HndWriteBarrier =  (inline caller) CoreCLR!GCHandleManager::StoreObjectInHandleIfNull+26
```
whereas before HndWriteBarrier was not inlined.

Reply:
Does it always get inlined on non-Windows? IIRC, we had performance bugs where we assumed small functions like this get inlinined without being in the header and marked as `inline`.

Reply:
resolving...please let me know if you don't agree. thanks!

Reply:
good point.. I don't :) will change.